src/libraries/System.Text.Json/tests/Common/RequiredKeywordTests.cs

```diff
@@ -568,6 +568,24 @@ public class ClassWithRequiredKeywordAndJsonRequiredCustomAttribute
             public required string SomeProperty { get; set; }
         }
 
+        [Fact]
+        public async Task ClassWithCustomRequiredPropertyName_Roundtrip()
+        {
+            // Regression test for https://github.com/dotnet/runtime/issues/82730
+            var value = new ClassWithCustomRequiredPropertyName { Property = 42 };
+            string json = await Serializer.SerializeWrapper(value);
+            Assert.Equal("""{"Prop":42}""", json);
+
+            value = await Serializer.DeserializeWrapper<ClassWithCustomRequiredPropertyName>(json);
+            Assert.Equal(42, value.Property);
+        }
+
+        public class ClassWithCustomRequiredPropertyName
+        {
+            [JsonPropertyName("Prop")]
+            public required int Property { get; set; }
```

Comment:
How about `required` + `init` since that's the exact scenario of the issue being fixed?

Reply:
Yeah I guess I can add that for completeness.

Reply:
I mentioned it for the sake of completeness. Up to you.

Reply:
No it's good feedback, we should be testing that scenario regardless. I've pushed changes to the `required` test.